conductor-client/src/main/kotlin/com/openlattice/organizations/WarehouseService.kt

```diff
@@ -0,0 +1,62 @@
+package com.openlattice.organizations
+
+import com.google.common.base.Preconditions
+import com.hazelcast.core.HazelcastInstance
+import com.openlattice.authorization.AuthorizationManager
+import com.openlattice.authorization.AclKey
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.authorization.HazelcastAclKeyReservationService
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.organizations.JdbcConnectionParameters
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+/**
+ * @author Andrew Carter andrew@openlattice.com
+ */
+
+@Service
+class WarehouseService(
+    hazelcastInstance: HazelcastInstance,
+    private val authorizationManager: AuthorizationManager,
+    private val aclKeyReservationService: HazelcastAclKeyReservationService
+) {
+
+    private val warehouses = HazelcastMap.WAREHOUSES.getMap(hazelcastInstance)
+
+    fun getWarehouses(): Iterable<JdbcConnectionParameters> {
+        return warehouses.values
+    }
+
+    fun getWarehouse(id: UUID): JdbcConnectionParameters {
+        return warehouses.getValue(id)
+    }
+
+    fun createWarehouse(jdbc: JdbcConnectionParameters): UUID {
+        val aclKey = reserveWarehouseIfNotExists(jdbc)
+        authorizationManager.setSecurableObjectType(aclKey, SecurableObjectType.JdbcConnectionParameters)
+        return jdbc._id
+    }
+
+    fun deleteWarehouse(id: UUID) {
+        ensureValidWarehouseId(id)
+        warehouses.delete(id)
+    }
+
+    fun updateWarehouse(jdbc: JdbcConnectionParameters): Int {
+        ensureValidWarehouseId(jdbc._id)
+        warehouses.replace(jdbc._id, jdbc)
+        return 1
```

Comment:
Is there a reason why you want to return an int here? I don't see the value. If the request returns with HTTP 200, then I know the update was successful.

Reply:
Ah yes that is redundant. I'll take that out.